biztown/src/components/FormInputLayout.js

```diff
@@ -1,260 +1,428 @@
-                freeResponseData[bankFormType][name] = value;
-                return { freeResponseData };
-            })
-        } else {
-            this.setState(prevState => {
-                var freeResponseData = JSON.parse(JSON.stringify(prevState.freeResponseData));
-                freeResponseData[bankFormType]["rowEntries"][name] = value;
-                return { freeResponseData };
-            })
+    var registerFormAnswerObj = {
+      q17_answer: `${registerFormData.balanceDollarAmount_0}.${registerFormData.balanceCentAmount_0}`,
+      q18_answer: `${registerFormData.balanceDollarAmount_1}.${registerFormData.balanceCentAmount_1}`,
+      q19_answer: registerFormData.transactionDesc_2,
+      q20_answer: `${registerFormData.paymentDollarAmount_2}.${registerFormData.paymentCentAmount_2}`,
+      q21_answer: `${registerFormData.balanceDollarAmount_2}.${registerFormData.balanceCentAmount_2}`,
+      q22_answer: `${registerFormData.balanceDollarAmount_3}.${registerFormData.balanceCentAmount_3}`,
+    };
+
+    // construct completed form from different objects
+    var completedForm = {
+      ...depositFormAnswerObj,
+      ...registerFormAnswerObj,
+      ...aboutSectionAnswerObj,
+      ...this.state.multipleChoiceData,
+      ...this.state.freeResponseData.checkSlip,
+      ...this.state.personalFinanceData,
+    };
+
+    console.log("completedForm", completedForm);
+
+    var submitPath = this.props.postTest
+      ? "postassessment/"
+      : "preassessment/";
+    DataManager.post(submitPath, completedForm).then(() => {
+      this.props.history.push("/completionpage");
+    });
+  };
+
+  validatePage = (pageObj) => {
+    for (let question of pageObj.keys) {
+      const element = document.getElementById(`${question}`);
+
+      if (document.getElementById(`${question}`).type === "radio") {
+        const section = this.state[`${pageObj.title}`];
+
+        if (!(element.name in section)) {
+          alert(`Please choose an option for ${question}.`);
+          return false;
         }
+      }
+
+      if (document.getElementById(`${question}`).value === "") {
+        const editedString =
+          question[0].toUpperCase() + question.substring(1);
+
+        alert(`${editedString} must be filled out.`);
+        return false;
+      }
     }
-    
-    setSchoolData = schoolData => {
-        this.setState({ schoolData });
-        // initialise to first school district
-        this.setInitialSchoolInfo(Object.keys(schoolData)[0]);
-    }
 
-    setInitialSchoolInfo = schoolDistrict => {
-        var schoolData = this.state.schoolData;
-        var aboutData = JSON.parse(JSON.stringify(this.state.aboutData));
-        var school = schoolData[schoolDistrict][0].id;
-        aboutData = { ...aboutData, schoolDistrict, school};
-        this.setState({ aboutData });
+    return true;
+  };
+
+  _next = (e) => {
+    // this.handleSubmit(e)
+
+    const aboutDataObj = {
```

Comment:
Tbh, not especially keen on the page objects being generated here. It seems a little clunky and a little too hard coded which makes me think that there is a better way of approaching it. I still need to think on it a bit more. Another approach here would be initialising these values to null in the state, but am not keen on that either. This is functional and fine, but worth thinking about.

Reply:
Alright, I think on this as well. Since it's currently functioning and they're pushing for MVP I'll leave it as is. I'll add it to the backlog.

Reply:
Since I've currently set all answers to be stored as strings, would initializing these values to empty strings the same or better than initializing them to null? If they're empty strings, it'll enable to write a conditional within validatePage() that doesn't rely on DOM elements.